lib/SetupChecks/MailConnectionPerformance.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * SPDX-FileCopyrightText: 2025 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+namespace OCA\Mail\SetupChecks;
+
+use OCA\Mail\Account;
+use OCA\Mail\Db\MailAccountMapper;
+use OCA\Mail\Db\ProvisioningMapper;
+use OCA\Mail\IMAP\FolderMapper;
+use OCA\Mail\IMAP\IMAPClientFactory;
+use OCP\IL10N;
+use OCP\SetupCheck\ISetupCheck;
+use OCP\SetupCheck\SetupResult;
+use Psr\Log\LoggerInterface;
+use Throwable;
+
+class MailConnectionPerformance implements ISetupCheck {
+	public function __construct(
+		private IL10N $l10n,
+		private LoggerInterface $logger,
+		private ProvisioningMapper $provisioningMapper,
+		private MailAccountMapper $accountMapper,
+		private IMAPClientFactory $clientFactory,
+		private FolderMapper $folderMapper,
+		private MicroTime $microtime,
+	) {
+	}
+
+	public function getName(): string {
+		return $this->l10n->t('mail connection performance');
```

Comment:
```suggestion
		return $this->l10n->t('Mail connection performance');
```

sorry, I wasn't precise. The start of the sting stays capitalized. Amend after your rebase+squash ;-)